### PR TITLE
Only try to lift boundary conditions if forms are provided

### PR DESCRIPTION
**Pull request by jpdean**
Resolves https://github.com/FEniCS/dolfinx/issues/3301

### CODE REVIEW

**Comment by nate-sime** [comment]
Should we add a test for this?

File: cpp/dolfinx/fem/assembler.h [hunk 1]
```diff
@@ -155,6 +155,10 @@ void apply_lifting(
         bcs1,
     const std::vector<std::span<const T>>& x0, T scale)
 {
+  // If all forms are null, there is nothing to do
+  if (std::all_of(a.begin(), a.end(), [](auto ptr) { return ptr == nullptr; }))
```
Comment (jhale) [inline cpp/dolfinx/fem/assembler.h:159]: You can use std::ranges::all_of here.

File: python/test/unit/fem/test_assembler.py [hunk 2]
```diff
@@ -1139,6 +1139,17 @@ def block():
...
 
+    def test_block_null_lifting(self):
+        from dolfinx.fem.petsc import assemble_vector_block
+
+        comm = MPI.COMM_WORLD
+        msh = create_unit_square(comm, 2, 2)
+        V = functionspace(msh, ("Lagrange", 1))
+        W = functionspace(msh, ("Lagrange", 2))
+        v, w = ufl.TestFunction(V), ufl.TestFunction(W)
+        L = form([v * ufl.dx, w * ufl.dx])
```
Comment (jorgensd) [inline python/test/unit/fem/test_assembler.py:1150]:
```suggestion
        L = form([ufl.conj(v) * ufl.dx, ufl.conj(w) * ufl.dx])
```
To support complex assembly
